lib/puppet/http/service.rb

```diff
@@ -31,6 +31,12 @@ def connect(ssl_context: nil)
     @client.connect(@url, ssl_context: ssl_context)
   end
 
+  def add_puppet_headers(headers)
+    modified_headers = headers.dup
+    modified_headers['X-Puppet-Profiling'] = 'true' if Puppet[:profile]
+    modified_headers
+  end
+
   protected
```

Comment:
Could you move `add_puppet_headers` below the `protected` line so that it's not `public`?

Reply:
done!